morango/syncsession.py

```diff
@@ -21,13 +21,23 @@ def _join_with_logical_operator(lst, operator):
     op = ") {operator} (".format(operator=operator)
     return "(({items}))".format(items=op.join(lst))
 
+def _get_server_ip(url):
+    try:
+        return socket.gethostbyname(url)
+    except:
+        return ''
 
-def _get_ip(url):
+def _get_client_ip_for_server(server_host, server_port):
+    server_port = server_port if server_port else 80
```

Comment:
More idiomatic would be just to set the default with `server_port=80` in the params. Except I don't think we should allow defaults here, let's just always be explicit. (It's not often going to be 80, more commonly 8080 or whatnot).

Ah, but I see that `urlparse` returns `None` and doesn't know to set the default to 80 based on the protocol. However, one thing to note is that the default port depends on http vs https.

Perhaps don't set a default here, and then change the calling code to: `_get_client_ip_for_server(hostname, url.port or (80 if url.scheme == "http" else 443))` (perhaps splitting the port extraction out into its own line)